website/public/js/controllers/partyCtrl.js

```diff
@@ -51,6 +51,11 @@ habitrpg.controller("PartyCtrl", ['$rootScope','$scope','Groups','Chat','User','
           Analytics.track({'hitType':'event','eventCategory':'button','eventAction':'click','eventLabel':'Leave Party'});
           $scope.selectedGroup = group;
           $scope.popoverEl = $($event.target);
+
+          if ($scope.popoverEl[0].localName == "span") {
+              $scope.popoverEl = $($scope.popoverEl[0].parentNode);
+          }
```

Comment:
Alright - sorry, I'm quite new to this! From what I've tested, this indeed also happens when clicking the leave button for the guilds:

![](http://i.imgur.com/4O3XCVS.png)

There's a TODO on the file suggesting the function to be refactored and used for both leaving parties and guilds. I suppose this would be a great time to do that - I'd be happy to do it but knowing so little about the structure of the code, I'd need some guidance on where the new method should be placed (is there a global JS file for this sort of thing?).

I went ahead and tested what @colwem suggested, along with some better code, and I came up with this - seems a lot better to me:

````
if (!$scope.popoverEl.hasClass("btn")) {
    $scope.popoverEl = $scope.popoverEl.parent();
}
````

What do you think?